AdminServer/appscale/admin/__init__.py

```diff
@@ -691,7 +700,47 @@ def identify_as_hoster(self, project_id, service_id, version):
       raise CustomHTTPError(
         HTTPCodes.INTERNAL_ERROR, message='Revision already exists')
 
-    # Remove old revision nodes.
+  def stop_hosting_revision(self, project_id, service_id, version):
+    """ Removes a revision and its hosting entry.
+
+    Args:
+      project_id: A string specifying a project ID.
+      service_id: A string specifying a service ID.
+      version: A dictionary containing version details.
+    """
+    revision_key = VERSION_PATH_SEPARATOR.join(
+      [project_id, service_id, version['id'], str(version['revision'])])
+    revision_node = '/apps/{}'.format(revision_key)
+    hoster_node = '/'.join([revision_node, options.private_ip])
```

Comment:
Stupid question, is this private_ip of the machine running the AdminServer? or is the list of private_ips we have in the deployment?

Reply:
It's the private IP of the machine running the AdminServer. The AdminServer claims that it has a revision archive by making an entry like `/apps/guestbook_default_v1_<revision>/<own_ip>`.